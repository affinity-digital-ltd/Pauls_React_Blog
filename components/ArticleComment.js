import { Component } from 'react'

import CommentForm from '../components/CommentForm'
import Auth from '../lib/Auth'

class ArticleComment extends Component {
  constructor (props) {
    super(props)

    this.auth = new Auth()
    this.login = this.login.bind(this)
    this.logout = this.logout.bind(this)
  }

  login () {
    this.auth.login()
  }

  userProfile () {
    return {
      name: localStorage.getItem('name')
    }
  }

  logout () {
    this.auth.logout()
  }

  render () {
    const userProfile = this.userProfile()
    if (this.auth.isAuthenticated()) {
      return <div>
        <div className='c-user-comments__info'>
          <h3 className='c-user-comments__login-message'>Welcome, {userProfile.name}</h3>
          <a className='c-user-comments__login-btn' onClick={this.logout}>Logout</a>
        </div>
        <CommentForm />
      </div>
    } else {
      return <div className='c-user-comments__info'>
        <h3 className='c-user-comments__login-message'>Please login to comment.</h3>
        <a className='c-user-comments__login-btn' onClick={this.login} >Login</a>
      </div>
    }
  }
}

export default ArticleComment
