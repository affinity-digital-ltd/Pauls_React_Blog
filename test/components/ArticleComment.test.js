import ArticleComment from '../../components/ArticleComment'
jest.mock('../../lib/Auth')

describe('<ArticleComment />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <ArticleComment />
    )

    expect(component).toMatchSnapshot()
  })
})