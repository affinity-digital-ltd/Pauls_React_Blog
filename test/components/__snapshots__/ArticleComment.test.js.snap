// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleComment /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleComment />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="c-user-comments__login-message"
        >
          Please login to comment.
        </h3>,
        <a
          className="c-user-comments__login-btn"
          onClick={[Function]}
        >
          Login
        </a>,
      ],
      "className": "c-user-comments__info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please login to comment.",
          "className": "c-user-comments__login-message",
        },
        "ref": null,
        "rendered": "Please login to comment.",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "className": "c-user-comments__login-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Login",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="c-user-comments__login-message"
          >
            Please login to comment.
          </h3>,
          <a
            className="c-user-comments__login-btn"
            onClick={[Function]}
          >
            Login
          </a>,
        ],
        "className": "c-user-comments__info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please login to comment.",
            "className": "c-user-comments__login-message",
          },
          "ref": null,
          "rendered": "Please login to comment.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "c-user-comments__login-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Login",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
